### DISCUSSION TITLE
'Restart on fail' is inconsistent between Sudden Death, Perfect and Accuracy Challenge

### DISCUSSION THREAD

**Discussion by tovc** [ppy/osu#31300] [Ideas] [answered]
The default setting for Sudden Death's 'Restart on fail' is disabled. The default setting for Perfect's 'Restart on fail' is enabled. The default setting for Accuracy Challenge's 'Restart on fail' is disabled.

'Restart on fail' should obviously be disabled for Perfect, but that's only half the issue. Sudden Death can be used in conjunction with Accuracy Challenge, which means that you can simultaneously have 'Restart on fail' enabled _and_ disabled. In this case, the enabled option is prioritised.
![image](https://github.com/user-attachments/assets/ac0f72bf-a3ca-4d75-b93b-71310959c220)

Ultimately, I think that removing 'Restart on fail' from all three mods is the best option here. The option has no bearing on the mod's function in submitted scores, so it could just be made into a separate mod, or moved entirely out of mods and into the game's options.

**Comment by Joehuu** [comment, marked as answer]
This is already tracked as an issue: https://github.com/ppy/osu/issues/28434

**Reply by tovc** [reply]
This issue is only partially tracked there. That issue refers to triggering based on which mod the fail was caused by. The inconsistent default states for the mods' options are not tracked there.

**Reply by Joehuu** [reply]
I understand the consistency issue if you're lazer only, but the Perfect mod being inconsistent is intentional to match osu!(stable).

**Reply by tovc** [reply]
Sudden Death/Perfect already don't match stable in terms of failing with Easy, so I don't see why they'd necessarily need to match it here. Sudden Death and Perfect are effectively options for the same mod in stable, and their functions are very similar, so the consistency feels like it shouldn't be there.

related: https://github.com/ppy/osu/discussions/13857
